include/vcpkg/base/curl.h

```diff
@@ -0,0 +1,17 @@
+
+#ifdef _MSC_VER
+#pragma warning(push)           // Save current warning state
+#pragma warning(disable : 6101) // Disable specific warning (e.g., warning 4996)
```

Comment:
Can you explain why 6101 specifically is suppressed here rather than explaining what pragma warning does?

Also this should use https://github.com/microsoft/vcpkg-tool/blob/1d7fdacf018b2e5cb4e679eaa3572b9ceb5f7a4b/include/vcpkg/base/pragmas.h#L29 rather than needing an explicit _MSC_VER guard. (See `base/fmt.h` for another example)